### PR TITLE
Doc: Rework Docker documentation

### PR DESCRIPTION
**Pull request by sgillespie**
# Description

Updates the docker documentation to (hopefully!) be more organized and up-to-date. Here is the [rendered version](https://github.com/sgillespie/cardano-db-sync/blob/doc/docker/doc/docker.md).

I've also created a new `docker-compose.example.yml`. The ideas around this are:

 1. Discourage usage in non-dev workflows
 2. Encourage users to maintain their own version
 3. Avoid tying this file to db-sync releases

To that end, I've left the existing one for now with the idea that we'll remove it in a few months.

# Checklist

- [x] Commit sequence broadly makes sense
- [x] Commits have useful messages
- [ ] New tests are added if needed and existing tests are updated
- [ ] Any changes are noted in the [changelog](https://github.com/IntersectMBO/cardano-db-sync/blob/master/db-sync/CHANGELOG.md)
- [ ] Code is formatted with [`fourmolu`](https://github.com/fourmolu/fourmolu) on version 0.10.1.0 (which can be run with `scripts/fourmolize.sh`)
- [x] Self-reviewed the diff

# Migrations

- [ ] The pr causes a [breaking change](https://github.com/IntersectMBO/cardano-db-sync/blob/master/doc/migrations.md) of type a,b or c
- [ ] If there is a breaking change, the pr includes a database migration and/or a fix process for old values, so that upgrade is possible
- [ ] Resyncing and running the migrations provided will result in the same database semantically

If there is a breaking change, especially a big one, please add a justification here. Please elaborate
more what the migration achieves, what it cannot achieve or why a migration is not possible.

### CODE REVIEW

File: doc/docker.md [hunk 1]
```diff
@@ -2,49 +2,57 @@
 
 ## Quickstart
 
-### Clone the repository
+We provide an example
+[docker-compose.yml](https://github.com/IntersectMBO/cardano-db-sync/blob/master/docker-compose.yml)
+to quickly get set up with `cardano-db-sync`.  Keep in mind that this is only a template,
```
Comment (Cmdv) [inline doc/docker.md:7]: double space in front of `  Keep`

File: doc/docker.md [hunk 2]
```diff
@@ -55,224 +63,211 @@ b73d677399de:/# psql -U postgres -c "SHOW ALL" | grep maintenance
...
+Specifies a `cardano-db-sync` snapshot archive to download and restore from. If omitted,
+it will sync from genesis.
+
+### `DB_SYNC_CONFIG` (optional)
+
+Overrides the `db-sync-config.json` provided by the network configuration.
+
+#### `EXTRA_DB_SYNC_ARGS` (optional)
+
+Extra command line arguments to pass to `cardano-db-sync`.
```
Comment (Cmdv) [inline doc/docker.md:110]: I'm not sure if you agree but would be cool to have little examples of each arg? Also maybe a little link to the command line args

File: doc/docker.md (hunk 1 above)
Comment (sgillespie) [reply doc/docker.md:7]: Sharp eye!

File: doc/docker.md (hunk 2 above)
Comment (sgillespie) [reply doc/docker.md:110]: > maybe a link to the command line args docs

On second thought, we apparently don't have CLI args docs anymore :shrug:

Comment (sgillespie) [reply doc/docker.md]: Fixed up a little bit, any better?

File: doc/docker.md (hunk 1 above)
Comment (Cmdv) [reply doc/docker.md]: 👓

File: doc/docker.md (hunk 2 above)
Comment (Cmdv) [reply doc/docker.md]: perfect ❤️ 

> On second thought, we apparently don't have CLI args docs anymore 🤷

did we/I accidentally remove them?

**Review by Cmdv — APPROVED** [review]
🚢 :shipit:

File: doc/docker.md (hunk 2 above)
Comment (sgillespie) [reply doc/docker.md]: No, we turned it into the config docs

Comment (Cmdv) [reply doc/docker.md]: ah yeah of course, I don't mind adding them in another PR so we get this PR through

**Pull request merged**